src/audio/base_fw.c

```diff
@@ -16,6 +16,14 @@
 #if defined(CONFIG_SOC_SERIES_INTEL_ACE)
 #include <intel_adsp_hda.h>
 #endif
+#include <sof/audio/module_adapter/module/generic.h>
+#include <sof/schedule/dp_schedule.h>
+#include <sof/schedule/ll_schedule.h>
+#ifdef CONFIG_SOF_TELEMETRY
+#include <sof/debug/telemetry/telemetry.h>
+#include "adsp_debug_window.h"
+#include "mem_window.h"
+#endif
```

Comment:
In case of telemetry.h , yes, you are right.
In case of other ones: I excluded some platforms from this patch, as there are some issues with memory window defines being already defined in platform configs.

Reply:
> I excluded some platforms from this patch, as there are some issues with memory window defines being already defined in platform configs.

So you are indeed "sweeping under the rug" exactly the sort of collision problem I was describing. This problem should at the very least be mentioned with a FIXME in the code, maybe pointing to a new bug. Maybe fixing this is less urgent than this PR, I don't know about priorities but you can't just **silently** hide collisions with a comment-less `#ifdef CONFIG_xxx #include` like this.

Kconfig's combinatorial explosion is always an untestable 
nightmare "by design"[*] and hiding problems is only making it worse for later.


[*] even more so considering kconfiglib.py does not implement "randconfig"

Reply:
Agreed.  What's being multiply defined here?  Maybe it's just a header missing an include guard?  As I read it this is only going to build on intel_adsp anyway, what are the other platforms in question?

Reply:
Yes, I understand. I mentioned the exclusion here on this page somewhere but adding fixme slipped my mind. I'll write something.

The problem is mainly on tgl. We have defines like  HP_SRAM_WIN0_BASE, HP_SRAM_WIN0_SIZE placed in src/platform/tigerlake/include/platform/lib/memory.h , for some reason.  I don't want to just remove them in case there is some context i don't know about. Other platforms get those defines from including the .h files mentioned here.